rpc/tests/utils/tiny_node.rs

```diff
@@ -0,0 +1,97 @@
+//! Tiny p2p node without any defined behaviour celestia can connect to so that we can test RPC p2p
+//! calls
+
+use celestia_types::p2p;
+use futures::StreamExt;
+use libp2p::{
+    core::upgrade::Version,
+    identify,
+    identity::{self, Keypair},
+    noise,
+    swarm::{keep_alive, NetworkBehaviour, SwarmBuilder, SwarmEvent},
+    tcp, yamux, Multiaddr, PeerId, Transport,
+};
+use tokio::{
+    task::JoinHandle,
+    time::{sleep, Duration},
+};
+
+// how long to wait during startup for node to start listening on interfaces, before we return a
+// list of addresses
+const NODE_ADDRESS_ACQUIRE_DELAY_TIME: Duration = Duration::from_millis(100);
+
+/// Our network behaviour.
+#[derive(NetworkBehaviour)]
+struct Behaviour {
+    identify: identify::Behaviour,
+    keep_alive: keep_alive::Behaviour,
+}
+
+impl Behaviour {
+    fn new(local_key: Keypair) -> Self {
+        let identify =
+            identify::Behaviour::new(identify::Config::new("".to_owned(), local_key.public()));
+
+        Self {
+            identify,
+            keep_alive: keep_alive::Behaviour,
+        }
+    }
+}
+
+async fn wait_for_addresses<T: NetworkBehaviour>(swarm: &mut libp2p::Swarm<T>) -> Vec<Multiaddr> {
+    let mut addresses = vec![];
+    let timeout = sleep(NODE_ADDRESS_ACQUIRE_DELAY_TIME);
+    tokio::pin!(timeout);
+
+    loop {
+        tokio::select! {
+            swarm_ev = swarm.select_next_some() => {
+                if let SwarmEvent::NewListenAddr { address, .. } = swarm_ev  {
+                    addresses.push(address);
+                }
+            },
+            () = &mut timeout => {
+                break;
+            }
+        }
+    }
+
+    addresses
+}
+
+pub async fn start_tiny_node() -> anyhow::Result<(p2p::AddrInfo, JoinHandle<()>)> {
```

Comment:
as in other comment, if this just spawns a task, we don't need a handle for it, it will get killed at the end of the test